mysql-test/r/ctype_utf16le.result

```diff
@@ -1168,6 +1168,196 @@ a
 2007-09-16
 DROP TABLE t1;
 #
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a datetime);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10-05:00');
+SELECT * FROM t1;
+a
+2016-12-10 18:10:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a datetime);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10+01:30');
+SELECT * FROM t1;
+a
+2016-12-10 11:40:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a datetime);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10Z');
+SELECT * FROM t1;
+a
+2016-12-10 13:10:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a timestamp);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10-05:00');
+SELECT * FROM t1;
+a
+2016-12-10 18:10:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a timestamp);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10+01:30');
+SELECT * FROM t1;
+a
+2016-12-10 11:40:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a timestamp);
+INSERT INTO t1 VALUES ('2016-12-10T10:10:10Z');
+SELECT * FROM t1;
+a
+2016-12-10 13:10:10
+DROP TABLE t1;
+#
+# Checking str_to_datetime() for RFC3339
+#
+select @@collation_connection;
+@@collation_connection
+utf16le_general_ci
+CREATE TABLE t1 (a datetime);
+INSERT INTO t1 VALUES ('1900-12-10T10:10:10Z');
+Warnings:
+Warning	1265	Data truncated for column 'a' at row 1
```

Comment:
This is warning because the date time is outside the Unix epoch range. The conversion is done using TIME_to_gmt_sec / gmt_sec_to_TIME. If the TIME_to_gmt_sec return an error, then the warning of MYSQL_TIME_WARN_TRUNCATED is set.